packages/SettingsProvider/src/com/android/providers/settings/SettingsProvider.java

```diff
@@ -2919,6 +2919,7 @@ public boolean ensureSettingsForUserLocked(int userId) {
             // Upgrade the settings to the latest version.
             UpgradeController upgrader = new UpgradeController(userId);
             upgrader.upgradeIfNeededLocked();
+            SettingsProviderHooks.insertSettingsOnBoot(SettingsRegistry.this, userId);
```

Comment:
This method is also called when user is added, correct? If so, the method name is misleading.